### PR TITLE
async/await

### PR DESCRIPTION
**Pull request by bscothern-Cricut**

### CODE REVIEW

File: Sources/FishyJoesCommonRuntime/Box.swift [hunk 1]
```diff
@@ -52,6 +52,14 @@ public struct Box<T> {
     public static func takeUnretainedOpaque(_ pointer: UnsafeMutableRawPointer) throws -> Box<T> {
         try Box(inner: AnyBox.takeUnretainedOpaque(pointer))
     }
+
+    public static func takeRetainedOpaque(_ pointer: UnsafeMutableRawPointer) throws -> Box<T> {
+        try Box(inner: AnyBox.takeRetainedOpaque(pointer))
+    }
+
+    public func callAsFunction<Argument, Result>(_ arg: Argument) throws -> Result where T == (Argument) throws -> Result {
```
Comment (cobbal) [inline Sources/FishyJoesCommonRuntime/Box.swift]: Can we put everything after == in parens?

File: Sources/FishyJoesCommonRuntime/Box.swift [hunk 2]
```diff
@@ -80,3 +88,52 @@ public final class AnyBox {
...
+
+    public static func takeUnretainedOpaque(_ pointer: UnsafeMutableRawPointer) throws -> Box<T> {
+        try Box(inner: AnyBox.takeUnretainedOpaque(pointer))
+    }
+
+    public static func takeRetainedOpaque(_ pointer: UnsafeMutableRawPointer) throws -> Box<T> {
+        try Box(inner: AnyBox.takeRetainedOpaque(pointer))
+    }
+
+    public func callAsFunction<Argument, Result>(_ arg: Argument) throws -> Result where T == (Argument) throws -> Result {
```
Comment (cobbal) [inline Sources/FishyJoesCommonRuntime/Box.swift]: and here

File: Sources/FishyJoesCore/Translators/KotlinTranslate.swift [hunk 3]
```diff
@@ -283,14 +312,19 @@ final class KotlinTranslator: Translator {
...
+
         return .method(
             KotlinClass.Method(
                 documentation: method.documentation,
                 isStatic: method.isStatic,
                 isOverride: method.exportAnnotation.isOverride,
                 name: exportAnnotation.name,
                 parameters: parameters,
-                returnType: context.resolve(type: method.returnType, generics: exportAnnotation.genericOverrides).kotlinType,
+                returnType: !method.isAsync ? context.resolve(type: method.returnType, generics: exportAnnotation.genericOverrides).kotlinType : .void,
```
Comment (cobbal) [inline Sources/FishyJoesCore/Translators/KotlinTranslate.swift]: Flip the order, lose the negation

File: Sources/FishyJoesCore/Translators/NodeTranslate.swift [hunk 4]
```diff
@@ -108,50 +108,97 @@ struct NodeTranslator: Translator {
...
+            return result + "converter: \(resolved.converterType.name).self)"
+        }
+
+        var taskBlock: (() -> Void) -> Void = { $0() }
+        if method.isAsync {
+            taskBlock = { body in
+                fragment.output("let (deferred, promise) = try env.env.createPromise()")
+                fragment.output("let envBox = UncheckedSendableBox(env.env)")
+
+                argIndex = 0
```
Comment (cobbal) [inline Sources/FishyJoesCore/Translators/NodeTranslate.swift]: Can this be made local? I don't like the idea of 2 loops using the same counter.

File: integration-tests/TestAPI-bindings/Package.swift [hunk 5]
```diff
@@ -46,6 +46,9 @@ let package = Package(
             path: "Sources/Generated/NodeInterface",
             resources: [
                 .copy("TestAPI.d.ts"),
+            ],
+            swiftSettings: [
+                .unsafeFlags(["-warn-concurrency"])
```
Comment (cobbal) [inline integration-tests/TestAPI-bindings/Package.swift:51]: Should this also be added to the template in FishyJoesExecute?

File: Sources/FishyJoesCore/Unparse/SourceFragment.swift [hunk 6]
```diff
@@ -49,7 +49,7 @@ class SourceFragment {
     }
 
     func outputBlock<R>(_ openingLine: String, closeWith close: String? = nil, newLineTerminated: Bool = true, semicolonTerminated: Bool = false, _ body: () throws -> R) rethrows -> R {
-        output(openingLine)
+        output(openingLine, newLineTerminated: !openingLine.isEmpty)
```
Comment (cobbal) [inline Sources/FishyJoesCore/Unparse/SourceFragment.swift]: Is this used? I don't like it.

File: integration-tests/TestAPI-bindings/node-test/package.json [hunk 7]
```diff
@@ -5,7 +5,7 @@
   "main": "index.js",
   "scripts": {
     "test": "npm run test-wasm && npm run test-node-native-macos",
-    "test-wasm": "node --experimental-vm-modules node_modules/jest/bin/jest.js --run-in-band --detect-open-handles --force-exit --debug --no-watchman -c jest.config.wasm.cjs",
+    "test-wasm": "node --inspect --experimental-vm-modules node_modules/jest/bin/jest.js --run-in-band --detect-open-handles --force-exit --debug --no-watchman -c jest.config.wasm.cjs",
```
Comment (cobbal) [inline integration-tests/TestAPI-bindings/node-test/package.json]: probably shouldn't commit this

**Review by cobbal — CHANGES_REQUESTED** [review]
Mostly minor stuff

File: Sources/FishyJoesCommonRuntime/Box.swift (hunk 1 above)
Comment (bscothern-Cricut) [reply Sources/FishyJoesCommonRuntime/Box.swift]: Done.

File: Sources/FishyJoesCore/Translators/KotlinTranslate.swift (hunk 3 above)
Comment (bscothern-Cricut) [reply Sources/FishyJoesCore/Translators/KotlinTranslate.swift]: Done. This should be temporary while moving to full async calls.

File: integration-tests/TestAPI-bindings/Package.swift (hunk 5 above)
Comment (bscothern-Cricut) [reply integration-tests/TestAPI-bindings/Package.swift:51]: Probably. We should discuss it but it makes sure we are taking note of how we are passing around data.

File: Sources/FishyJoesCore/Unparse/SourceFragment.swift (hunk 6 above)
Comment (bscothern-Cricut) [reply Sources/FishyJoesCore/Unparse/SourceFragment.swift]: It isn't being used anymore. It was an early aspect of how I was generating some stuff. I will remove it.

File: integration-tests/TestAPI-bindings/node-test/package.json (hunk 7 above)
Comment (bscothern-Cricut) [reply integration-tests/TestAPI-bindings/node-test/package.json]: Good catch. Removed.

File: Sources/FishyJoesCore/Translators/NodeTranslate.swift (hunk 4 above)
Comment (bscothern-Cricut) [reply Sources/FishyJoesCore/Translators/NodeTranslate.swift]: You use it for several things. I'll just change mine to be a local so the behavior in the other spots aren't changed.

File: Sources/FishyJoesCommonRuntime/Box.swift (hunk 2 above)
Comment (bscothern-Cricut) [reply Sources/FishyJoesCommonRuntime/Box.swift]: Fixed.

**Pull request merged**